rust/src/utils/nom.rs

```
@@ -0,0 +1,71 @@
+		use nom::combinator::cut;
+
+		// Since `cut` takes its argument by value,
+		// `cut`ting the parsers inside the closure makes it `FnOnce`
+		// (it can only be moved once inside the closure,
+		// where it can only be moved once to `cut`).
+		//
+		// Constructing the `cut` parsers outside,
+		// the closure is `FnMut` is `Parser::parse` takes `&mut self`.
+		//
+		// `cut` returns `impl Parser`, so we can't annotate with an explicit type.
+		let mut cut_inside: _ = cut(inside);
```

Clippy (1.96.0)

[warning] variable declared with type underscore

(warning)